content/docs/reactivesearch/react-searchbox/apireference.md

```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
--   **enableAppbase** `boolean` [optional]
-    enableAppbase is false by default. You can set this to true when you're using appbase.io alongside Elasticsearch. It enables the following features:
-    -   Recording of analytics events - search and clicks. [Read more](/docs/reactivesearch/v3/advanced/analytics/).
-    -   Query generation happens on server side - protecting against security concerns around query injection.
-    -   Apply query rules and functions for search queries. [Read more](/docs/search/rules/).
-    -   Apply additional security controls to requests: authenticate via RBAC (via JWTs) or Basic Auth, ACL based access control, IP based rate limits, IP/HTTP Referers whitelisting, fields filtering. [Read more](/docs/security/role/).
--   **enableQuerySuggestions** `bool` [optional]
+
+-   **appbaseConfig** `Object` [optional]
+    This allows you to customize the analytics experience when appbase.io is used as a backend. It accepts an object which has the following properties:
+
+    -   **recordAnalytics** Boolean allows recording search analytics (and click analytics) when set to true and appbase.io is used as a backend. Defaults to false.
+
+    -   **enableQueryRules** Boolean If false, then appbase.io will not apply the query rules on the search requests. Defaults to true.
+
+    -   **userId** String It allows you to define the user id to be used to record the appbase.io analytics. Defaults to the client's IP address.
+
+    -   **customEvents** Object It allows you to set the custom events which can be used to build your own analytics on top of appbase.io analytics. Further, these events can be used to filter the analytics stats from the appbase.io dashboard.
+
+-   **headers** `Object` [optional] set custom headers to be sent with each server request as key/value pairs.
+
+-   **transformRequest** `Function` Enables transformation of network request before
+    execution. This function will give you the request object as the param and expect an updated request in return, for execution.
+
+-   **transformResponse** `Function` Enables transformation of search network response before  
+    rendering them. It is an asynchronous function which will accept an Elasticsearch response object as param and is expected to return an updated response as the return value.
+
+    > Note: transformResponse function is expected to return data in the following structure.
+
+    ```javascript
+         {
+        // Elasticsearch hits response
+        hits: {
+            hits: [...],
+            total: 100
+        },
+        // Elasticsearch aggregations response
+        aggregations: {
+
+        }
+        took: 1
+    }
+    ```
+
+## SearchBox
+
+The core of the library which helps with data access and manipulation with any UI components
+
+### Props
```

Comment:
@nikhil-varma Can you break props into sections?
Ref. `SearchBase` docs PR
Also, add examples for non-intuitive props, for e.g `defaultSuggestions`.